frontend/src/pages/dashboard/profile/Profile.tsx

```diff
@@ -1,189 +1,83 @@
-			</div>
-			
-			<hr/>
-			<h3>Basic Information</h3>
-			<table style={ { width: "100%" } }>
-				<tbody>
-				<tr>
-					<td style={ { width: "50%", paddingRight: "calc(var(--spacing) / 2)" } }>
-						<label className={ "caption" } htmlFor={ "firstname" }>First Name</label>
-					</td>
-					<td style={ { width: "50%", paddingLeft: "calc(var(--spacing) / 2)" } }>
-						<label className={ "caption" } htmlFor={ "lastname" }>Last Name</label>
-					</td>
-				</tr>
-				<tr>
-					<td style={ { paddingRight: "calc(var(--spacing) / 2)" } }>
-						<input type={ "text" } className={ "form-input" } id={ "firstname" }
-							   placeholder={ "John" } defaultValue={ "Benni" } style={ { width: "100%" } }/>
-					</td>
-					<td style={ { paddingLeft: "calc(var(--spacing) / 2)" } }>
-						<input type={ "text" } className={ "form-input" } id={ "lastname" }
-							   placeholder={ "Doe" } defaultValue={ "Loidl" } style={ { width: "100%" } }/>
-					</td>
-				</tr>
-				<tr>
-					<td style={ { paddingRight: "calc(var(--spacing) / 2)" } }>
-						<label className={ "caption" } htmlFor={ "email" }>Email</label>
-					</td>
-				</tr>
-				<tr>
-					<td style={ { paddingRight: "calc(var(--spacing) / 2)" } }>
-						<input type={ "email" } className={ "form-input" } id={ "email" }
-							   placeholder={ "john.doe@gmail.com" } style={ { width: "100%" } }/>
-					</td>
-				</tr>
-				</tbody>
-			</table>
-		</div>
-		
-		<div className={ "container profile-expert-topics" }>
-			<h2>
-				<i className={ "fi fi-rr-brain" }/>
-				Expert Topics
-			</h2>
-			<p>Choose your expert topics to help others find you.</p>
-			
-			<hr/>
-			
-			<div style={ { display: "flex", gap: "var(--spacing)" } }>
-				<div style={ { flex: 1 } }>
-					<h3>My Expert Topics</h3>
-					<p>Choose your expert topics to help others find you.</p>
-				</div>
-				
-				<div style={ { flex: 1 } }>
-					<h3>Choose Expert Topics</h3>
-					<p>Choose your expert topics to help others find you.</p>
-				</div>
+					: <h1><Skeleton width={ 400 }/></h1>
+				}
+				{ profile && <Button icon={ "fi fi-rr-refresh" }
+                                     disabled={ updateUsernameInputValue === profile.name || updateUsernameInputValue.length < 4 || updateUsernameInputValue.length > 20 }
+                                     onClick={ async () => {
+										 await global.axios.post("/profile/update", { name: updateUsernameInputValue }, { withCredentials: true })
+													 .then(res => {
+														 setProfile({
+															 id: res.data.userId ?? "",
+															 name: res.data.username ?? "",
+															 type: res.data.accountState ?? "",
+															 registrationDate: res.data.registrationDate ?? ""
+														 });
+													 })
+													 .catch(err => axiosError(err, alert));
+									 } }>
+                    Update
+                </Button> }
 			</div>
-		</div>
-		
-		<div className={ "container profile-memberships" }>
```

Comment:
why is the functionality to upgrade to pro user deleted?
I know the functionality to upgrade with payment is not implemented in backend, but there should be the possibility to do so in frontend.

Reply:
There has never been upgrade functionality.